### DISCUSSION TITLE
Enhancement of Metrics for Harbor

### DISCUSSION THREAD

**Discussion by ninjadq** [goharbor/harbor#14419] [Ideas]
We can discuss here what metrics you wanna add in future release

**Comment by ruster-cn** [comment]
Whether the job service module should expose some indicators for processing, such as the number of pending tasks, task processing time and so on. If you configure a large number of webhooks, you are worried about the delay of webhook.

**Reply by ruster-cn** [reply]
https://github.com/goharbor/harbor/issues/14541 ，Similar to this kind of problem, if there is monitoring of task failure, the positioning will be faster

**Reply by ninjadq** [reply]
Yes, we would add pending queue size metrics on jobservice

**Comment by ruster-cn** [comment]
Is there a way to count the success and failure indicators of image pull? This indicator is actually more important. It should be included in the SLA of image warehouse. At present, we collect information from the perspective of users. It would be better if the server could also have one

**Reply by ninjadq** [reply]
did you tried use return code like `500` in API to track the error of pull
